components/core/src/clp/streaming_archive/ArchiveMetadata.cpp

```diff
@@ -22,15 +29,27 @@ ArchiveMetadata::ArchiveMetadata(
                          + sizeof(m_end_timestamp) + sizeof(m_compressed_size);
 }
 
-ArchiveMetadata::ArchiveMetadata(FileReader& file_reader) {
-    file_reader.read_numeric_value(m_archive_format_version, false);
-    file_reader.read_numeric_value(m_creator_id_len, false);
-    file_reader.read_string(m_creator_id_len, m_creator_id, false);
-    file_reader.read_numeric_value(m_creation_idx, false);
-    file_reader.read_numeric_value(m_uncompressed_size, false);
-    file_reader.read_numeric_value(m_compressed_size, false);
-    file_reader.read_numeric_value(m_begin_timestamp, false);
-    file_reader.read_numeric_value(m_end_timestamp, false);
+auto ArchiveMetadata::create_from_file(std::string_view path) -> ArchiveMetadata {
+    auto file_reader = FileReader(std::string(path));
+    struct stat file_stat{};
+    if (auto const clp_rc = file_reader.try_fstat(file_stat);
+        clp::ErrorCode::ErrorCode_Success != clp_rc)
+    {
+        throw OperationFailed(clp_rc, __FILENAME__, __LINE__);
+    }
+
+    clp::Array<char> buf(file_stat.st_size);
```

Comment:
```suggestion
    clp::Array<char> buf{static_cast<size_t>(file_stat.st_size)};
```

* We should use brace init.
* Brace init results in a compilation error due to the implicit conversion (precisely what brace init is meant to catch) forcing us to explicit convert the value.